tailfs/tailfsimpl/compositedav/stat_cache_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package compositedav
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"tailscale.com/tstest"
+)
+
+func TestStatCache(t *testing.T) {
+	// Make sure we don't leak goroutines
+	tstest.ResourceCheck(t)
+
+	val := []byte("1")
+	ttl := 1 * time.Second
+	c := &StatCache{TTL: ttl}
+
+	// set new stat
+	c.set("file", 1, val)
+	fetched := c.get("file", 1)
+	if !bytes.Equal(fetched, val) {
+		t.Errorf("want %q, got %q", val, fetched)
+	}
+
+	// fetch stat again, should still be cached
+	fetched = c.get("file", 1)
+	if !bytes.Equal(fetched, val) {
+		t.Errorf("want %q, got %q", val, fetched)
+	}
+
+	// wait for cache to expire and refetch stat, should be empty now
+	time.Sleep(ttl * 2)
+
+	fetched = c.get("file", 1)
+	if fetched != nil {
+		t.Errorf("invalidate should have cleared cached value")
+	}
+
+	c.set("file", 1, val)
```

Comment:
To reduce the chance of bad CI tripping on flakes, I suggest splitting these into two subtests: One with a long TTL (so that invalidate is more certain to happen before the TTL) and one with a short TTL (so that the wait for expiry can be longer).

I know these times seem ridiculously generous, but GitHub actions runners are amazingly good at triggering weird timing behaviours in tests 😆